src/main/java/uk/ac/ebi/eva/pipeline/listeners/StepProgressListener.java

```diff
@@ -34,13 +35,25 @@ public void beforeChunk(ChunkContext context) {
 
     @Override
     public void afterChunk(ChunkContext context) {
-        logger.info("Chunk stats: Items read count {}, items write count {}, items skip count{}",
-                context.getStepContext().getStepExecution().getReadCount(),
-                context.getStepContext().getStepExecution().getWriteCount(),
-                context.getStepContext().getStepExecution().getReadSkipCount()
-                        + context.getStepContext().getStepExecution().getProcessSkipCount()
-                        + context.getStepContext().getStepExecution().getWriteSkipCount()
-        );
+
+        int estimatedTotalNumberOfLines = (int)context.getStepContext().getStepExecutionContext()
+                .get(JobParametersNames.NUMBER_OF_LINES);
+
+        int read = context.getStepContext().getStepExecution().getReadCount();
+        int write = context.getStepContext().getStepExecution().getWriteCount();
+        int skip = context.getStepContext().getStepExecution().getReadSkipCount()
+                + context.getStepContext().getStepExecution().getProcessSkipCount()
+                + context.getStepContext().getStepExecution().getWriteSkipCount();
+
+        String chunkStatisticsMessage = "Chunk stats: Items read count " + read + ", items write count " + write + " items skip count " + skip;
+
+        if (estimatedTotalNumberOfLines != 0) {
+            int percent = (read * 100) / estimatedTotalNumberOfLines;
+            logger.info(percent + "% " + chunkStatisticsMessage);
+        } else {
+            logger.info(chunkStatisticsMessage);
+        }
```

Comment:
I can change this but I think it is more readable without the builder...
https://dzone.com/articles/string-concatenation-performacne-improvement-in-ja